### ISSUE TITLE
showAllValues undismissable on iOS

### ISSUE THREAD

**Issue by dracos** [alphagov/accessible-autocomplete#185]
Turning off showAllValues, it is okay (in that it appears when you type and disappears when you delete all the text).

![ios dropdown](https://user-images.githubusercontent.com/4776/27921248-12a8644e-626f-11e7-8af6-2fb72016ec09.gif)

**Comment by tvararu** [comment]
Thanks for the report!

**Comment by dracos** [comment]
Really want to get that out of the door, but will come back and have a look in a sec!

**Comment by edwardhorsford** [comment]
@tvararu is this related to us wanting the menu to remain visible when the keyboard is dismissed?

**Comment by dracos** [comment]
Right, now that https://www.fixmystreet.com/reports is live and using this autocomplete for the council select… :)

Yes, it is caused because `keepMenuOpen` is true if it's open on an iOS device. This was introduced in 6f8c141e15227c3f53b34972f15f44e45263a80a . So no blur will remove the menu, and there's no way to remove it at all. Not sure how you'd want to fix it – close it if it's open and there's a click/touch anywhere else?

**Comment by tvararu** [comment]
@edwardhorsford yes, it is, but as @dracos has pointed out there was a way to dismiss it pre-`showAllValue`: you could delete all the inputted text.

This is tricky. `blur` on the input field gets triggered on iOS when you close the on-screen keyboard. That's always been something we've implemented against, as the on-screen keyboard makes seeing the available options more difficult.

After doing more accessibility testing I found that `blur` on the individual options is handled differently on iOS if VoiceOver is on.

The only solution I could find at the time was to prevent dismissal of the menu for all situations but the one where the user selects something or deletes the content of the input. With `showAllValues`, one of those methods no longer works, so users are forced to choose something.

> Not sure how you'd want to fix it – close it if it's open and there's a click/touch anywhere else?

There are a few more issues that could be solved by adding a click handler on the parent window:

- https://github.com/alphagov/accessible-autocomplete/issues/48
- https://github.com/alphagov/accessible-autocomplete/issues/61

So it looks like we might have to go this route; we have to reimplement `blur`.

**Comment by edwardhorsford** [comment]
Yes, ideally clicking outside the menu should dismiss - this mimics the behaviour of other modals and dialogs.

I think we could also insert a close action (X) in the input when the menu is open. I know some browsers already do this - is there a way for us to override that and add our own?
